diff --git a/fhi_lib/lib/src/xran_common.c b/fhi_lib/lib/src/xran_common.c
index 5bbacf7..270ee17 100644
--- a/fhi_lib/lib/src/xran_common.c
+++ b/fhi_lib/lib/src/xran_common.c
@@ -256,8 +256,9 @@ int process_mbuf(struct rte_mbuf *pkt)
             print_dbg("Transport layer fragmentation (eCPRI) is not supported\n");
         }
 
-    } else if (Ant_ID >= p_x_ctx->PrachCPConfig.eAxC_offset && p_x_ctx->fh_init.prachEnable) {
+    } else if (Ant_ID+1 >= p_x_ctx->PrachCPConfig.eAxC_offset && p_x_ctx->fh_init.prachEnable) {
         /* PRACH packet has ruportid = num_eAxc + ant_id */
+        Ant_ID = Ant_ID + 1;
         Ant_ID -= p_x_ctx->PrachCPConfig.eAxC_offset;
         symbol_total_bytes += num_bytes;
         if (seq.e_bit == 1) {
diff --git a/fhi_lib/lib/src/xran_main.c b/fhi_lib/lib/src/xran_main.c
index 3b187fc..ef423d7 100644
--- a/fhi_lib/lib/src/xran_main.c
+++ b/fhi_lib/lib/src/xran_main.c
@@ -412,7 +412,8 @@ int xran_init_prach(struct xran_fh_config* pConf, struct xran_device_ctx * p_xra
         printf("PRACH start symbol %u lastsymbol %u\n", p_xran_dev_ctx->prach_start_symbol[0], p_xran_dev_ctx->prach_last_symbol[0]);
     }
 
-    pPrachCPConfig->eAxC_offset = xran_get_num_eAxc(NULL);
+    //pPrachCPConfig->eAxC_offset = xran_get_num_eAxc(NULL);
+    pPrachCPConfig->eAxC_offset = 5;
     print_dbg("PRACH eAxC_offset %d\n",  pPrachCPConfig->eAxC_offset);
 
     return (XRAN_STATUS_SUCCESS);
@@ -1036,7 +1037,7 @@ int xran_cp_create_and_send_section(void *pHandle, uint8_t ru_port_id, int dir,
         params.numSections          = 1;//nsection;
         params.sections             = sect_geninfo;
 
-        ret = xran_prepare_ctrl_pkt(mbuf, &params, cc_id, ru_port_id, seq_id);
+        ret = xran_prepare_ctrl_pkt(mbuf, &params, cc_id, ru_port_id+1, seq_id);
         if(ret < 0) {
             print_err("Fail to build control plane packet - [%d:%d:%d] dir=%d\n",
                         frame_id, subframe_id, slot_id, dir);
@@ -1269,7 +1270,8 @@ void tx_cp_ul_cb(struct rte_timer *tim, void *arg)
                         struct xran_cp_gen_params params;
                         struct xran_section_gen_info sect_geninfo[8];
                         struct rte_mbuf *mbuf = xran_ethdi_mbuf_alloc();
-                        prach_port_id = ant_id + num_eAxc;
+                        //prach_port_id = ant_id + num_eAxc;
+												prach_port_id = 5;
                         /* start new section information list */
                         xran_cp_reset_section_info(pHandle, XRAN_DIR_UL, cc_id, prach_port_id, ctx_id);
 
diff --git a/fhi_lib/lib/src/xran_up_api.c b/fhi_lib/lib/src/xran_up_api.c
index a69712f..3fcea46 100644
--- a/fhi_lib/lib/src/xran_up_api.c
+++ b/fhi_lib/lib/src/xran_up_api.c
@@ -131,7 +131,8 @@ static int xran_build_ecpri_hdr_ex(struct rte_mbuf *mbuf,
     ecpri_hdr->cmnhdr.ecpri_payl_size = rte_cpu_to_be_16(ecpri_payl_size);
 
     /* one to one lls-CU to RU only and band sector is the same */
-    ecpri_hdr->ecpri_xtc_id = xran_compose_cid(0, 0, CC_ID, Ant_ID);
+    //ecpri_hdr->ecpri_xtc_id = xran_compose_cid(0, 0, CC_ID, Ant_ID);
+    ecpri_hdr->ecpri_xtc_id = xran_compose_cid(0, 0, CC_ID, 1);
 
     ecpri_hdr->ecpri_seq_id.seq_id = seq_id;
 
@@ -362,7 +363,7 @@ int32_t xran_extract_iq_samples(struct rte_mbuf *mbuf,
     xran_decompose_cid((uint16_t)ecpri_hdr->ecpri_xtc_id, &result);
 
     *CC_ID  = result.ccId;
-    *Ant_ID = result.ruPortId;
+    *Ant_ID = result.ruPortId - 1;
 
     /* Process radio header. */
     struct radio_app_common_hdr *radio_hdr =
-- 
2.25.1

